Fix http request memory leak

EricDahlvang opened the pull request:
Fixes https://github.com/microsoft/botbuilder-dotnet/issues/5261

gabog commented on libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/HttpRequest.cs:
```diff
@@ -228,187 +228,196 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
+                            {
+                                traceInfo.request.content = instanceBody.ToString();
+                                traceInfo.request.headers = JObject.FromObject(putContent.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
+                                request.Content = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType);
```
Isn't this the putContent we created in the using in line 332? Shouldn't we be using that disposable instance instead of creating a new instance of StringContent?

gabog commented on libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/HttpRequest.cs:
```diff
@@ -228,187 +228,196 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
+                            {
+                                traceInfo.request.content = instanceBody.ToString();
+                                traceInfo.request.headers = JObject.FromObject(postContent?.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
+                                request.Content = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType);
```
Isn't this the postContent we created in the using in line 299? Shouldn't we be using that disposable instance instead of creating a new instance of StringContent?

gabog reviewed (changes requested):
Check the comments around StringContent usage for PUT and POST.
The rest are style changes that you can apply if time allows.

EricDahlvang commented:
Thank you for reviewing @gabog 

Unfortunately, this PR was towards 4.12 and we've decided to not take this in 4.12.  So, I've addressed your feedback in a new PR targeting the main branch: https://github.com/microsoft/botbuilder-dotnet/pull/5268
